Inferences/ArithmeticSubtermGeneralization/AdditionGeneralizationImpl.cpp

```diff
@@ -24,7 +24,7 @@ namespace AdditionGeneralizationImpl {
 
 template<class NumTraits> class MonomSet;
 
-using GenMap = Map<Variable, AnyNumber<MonomSet>>;
+using GenMap = Map<Variable, AnyNumber<MonomSet>, StlHash>;
```

Comment:
Anywhere where `StlHash` was inserted: this used to be resolving to the default hash functions, which might do all manner of crazy. Since for all these types there is some kind of `std::hash` implementation, I assume this was meant to use that.